src/lib/bootstrap_controller/bootstrap_controller.ml

```diff
@@ -54,13 +64,22 @@ let time_deferred deferred =
   (Time.diff end_time start_time, result)
 
 let worth_getting_root t candidate =
```

Comment:
In cases like this, shouldn't we be sticking the context into `t` instead of unpacking the values from `t` into a context?